src/test/ui/feature-gate/issue-43106-gating-of-test.stderr

```diff
@@ -1,7 +1,13 @@
+error: malformed `test` attribute input
+  --> $DIR/issue-43106-gating-of-test.rs:10:1
+   |
+LL | #![test                    = "4200"]
```

Comment:
Did we previously silently allow this? If so we may want to do this in a separate PR since the age of `#[test]` means that this could be likely for causing accidental regressions.

Reply:
Yes, this was previously allowed.
I don't want to wait for a separate crater run for this, since I don't expect *this* kind of misuse to be common after seeing the statistics from https://github.com/rust-lang/rust/pull/57321.

I think I'll just demote this error to [this lint](https://github.com/rust-lang/rust/issues/57571) for `#[test]` and `#[bench]`.